src/torchcodec/_core/custom_ops.cpp

```diff
@@ -796,72 +796,169 @@ std::string get_stream_json_metadata(
-    map["bitRate"] = std::to_string(*streamMetadata.bitRate);
-  }
-  if (streamMetadata.numFramesFromContent.has_value()) {
-    map["numFramesFromContent"] =
-        std::to_string(*streamMetadata.numFramesFromContent);
-  }
-  if (streamMetadata.numFramesFromHeader.has_value()) {
-    map["numFramesFromHeader"] =
-        std::to_string(*streamMetadata.numFramesFromHeader);
-  }
-  if (streamMetadata.beginStreamSecondsFromHeader.has_value()) {
-    map["beginStreamSecondsFromHeader"] =
-        std::to_string(*streamMetadata.beginStreamSecondsFromHeader);
-  }
-  if (streamMetadata.beginStreamPtsSecondsFromContent.has_value()) {
-    map["beginStreamSecondsFromContent"] =
-        std::to_string(*streamMetadata.beginStreamPtsSecondsFromContent);
-  }
-  if (streamMetadata.endStreamPtsSecondsFromContent.has_value()) {
-    map["endStreamSecondsFromContent"] =
-        std::to_string(*streamMetadata.endStreamPtsSecondsFromContent);
-  }
-  if (streamMetadata.codecName.has_value()) {
-    map["codec"] = quoteValue(streamMetadata.codecName.value());
-  }
-  if (streamMetadata.width.has_value()) {
-    map["width"] = std::to_string(*streamMetadata.width);
-  }
-  if (streamMetadata.height.has_value()) {
-    map["height"] = std::to_string(*streamMetadata.height);
-  }
-  if (streamMetadata.sampleAspectRatio.has_value()) {
-    map["sampleAspectRatioNum"] =
-        std::to_string((*streamMetadata.sampleAspectRatio).num);
-    map["sampleAspectRatioDen"] =
-        std::to_string((*streamMetadata.sampleAspectRatio).den);
-  }
-  if (streamMetadata.averageFpsFromHeader.has_value()) {
-    map["averageFpsFromHeader"] =
-        std::to_string(*streamMetadata.averageFpsFromHeader);
-  }
-  if (streamMetadata.sampleRate.has_value()) {
-    map["sampleRate"] = std::to_string(*streamMetadata.sampleRate);
-  }
-  if (streamMetadata.numChannels.has_value()) {
-    map["numChannels"] = std::to_string(*streamMetadata.numChannels);
-  }
-  if (streamMetadata.sampleFormat.has_value()) {
-    map["sampleFormat"] = quoteValue(streamMetadata.sampleFormat.value());
-  }
-  if (streamMetadata.mediaType == AVMEDIA_TYPE_VIDEO) {
-    map["mediaType"] = quoteValue("video");
-  } else if (streamMetadata.mediaType == AVMEDIA_TYPE_AUDIO) {
-    map["mediaType"] = quoteValue("audio");
+  // Check whether content-based metadata is available for this stream.
+  // In exact mode: content-based metadata exists for all streams.
+  // In approximate mode: content-based metadata does not exist for any stream.
+  // In custom_frame_mappings: content-based metadata exists only for the active
+  // stream.
+  // Our fallback logic assumes content-based metadata is available.
+  // It is available for decoding on the active stream, but would break
+  // when getting metadata from non-active streams.
+  if ((seekMode != SeekMode::custom_frame_mappings) ||
+      (seekMode == SeekMode::custom_frame_mappings &&
+       stream_index == activeStreamIndex)) {
```

Comment:
We should be able to factor out all of values that are common to both cases, such as `durationSecondsFromHeader` and `mediaType`. That way, it's more clear what the difference between the two cases is.